pkg/utils/locking.go

```diff
@@ -77,7 +92,9 @@ func (projectLock *ProjectLockImpl) Lock(prNumber int) (bool, error) {
 		return false, err
 	}
 
-	if lockAcquired {
+	_, isNoOpLock := projectLock.InternalLock.(*NoOpLock)
```

Comment:
I think this kinda defeats the purpose of having polymorphic lock if we need to do a type check.

Reply:
Thank you for your feedback. Regarding the `NoOpLock`, would it be possible to add an additional return parameter called `lockIgnored` to the `Lock` interface instead of using this alternative approach?
I'm open to exploring other ideas as well. Please let me know if you have any suggestions or recommendations.